System.Device.I2c/I2cDevice.cs

```diff
@@ -11,25 +11,40 @@ namespace System.Device.I2c
     /// </summary>
     public class I2cDevice : IDisposable
     {
+        // this is used as the lock object 
+        // a lock is required because multiple threads can access the device
+        [Diagnostics.DebuggerBrowsable(Diagnostics.DebuggerBrowsableState.Never)]
+        private readonly object _syncLock;
 
-        private Windows.Devices.I2c.I2cDevice _device;
-        // For the ReadByte and WriteByte operations
-        private byte[] bufferSingleOperation = new byte[1];
+        [Diagnostics.DebuggerBrowsable(Diagnostics.DebuggerBrowsableState.Never)]
+        private readonly I2cConnectionSettings _connectionSettings;
+
+        [Diagnostics.DebuggerBrowsable(Diagnostics.DebuggerBrowsableState.Never)]
+        private bool _disposed;
+
+        // speeds up the execution of ReadByte and WriteByte operations
+        private readonly byte[] _buffer;
 
         /// <summary>
         /// The connection settings of a device on an I2C bus. The connection settings are immutable after the device is created
         /// so the object returned will be a clone of the settings object.
         /// </summary>
-        public I2cConnectionSettings ConnectionSettings { get; }
+        public I2cConnectionSettings ConnectionSettings { get => _connectionSettings; } 
 
         /// <summary>
         /// Reads a byte from the I2C device.
         /// </summary>
         /// <returns>A byte read from the I2C device.</returns>
         public byte ReadByte()
         {
-            _device.Read(bufferSingleOperation);
-            return bufferSingleOperation[0];
+            lock (_syncLock)
+            {
+                var buffer = new SpanByte(_buffer);
```

Comment:
this is not needed as _buffer should be a SpanByte directly

Reply:
Can't do...
CS8345	Field or auto-implemented property cannot be of type 'SpanByte' unless it is an instance member of a ref struct.